src/instance.rs

```diff
@@ -836,18 +832,18 @@ where
 
 impl<T> Py<T> {
     /// Attaches this `Py` to the given Python context, allowing access to further Python APIs.
-    pub(crate) fn attach<'py>(&self, _py: Python<'py>) -> &Py2<'py, T> {
-        // Safety: `Py2` has the same layout as `Py`
+    pub(crate) fn attach<'py>(&self, _py: Python<'py>) -> &Bound<'py, T> {
+        // Safety: `Bound` has the same layout as `Py`
         unsafe { &*(self as *const Py<T>).cast() }
     }
 
     /// Same as `attach` but takes ownership of `self`.
-    pub(crate) fn attach_into(self, py: Python<'_>) -> Py2<'_, T> {
-        Py2(py, ManuallyDrop::new(self))
+    pub(crate) fn attach_into(self, py: Python<'_>) -> Bound<'_, T> {
+        Bound(py, ManuallyDrop::new(self))
     }
 
-    pub(crate) fn attach_borrow<'a, 'py>(&'a self, py: Python<'py>) -> Py2Borrowed<'a, 'py, T> {
-        Py2Borrowed(self.0, PhantomData, py)
+    pub(crate) fn attach_borrow<'a, 'py>(&'a self, py: Python<'py>) -> Borrowed<'a, 'py, T> {
+        Borrowed(self.0, PhantomData, py)
```

Comment:
I think these three methods should probably rename, maybe `attach` -> `bind`, `attach_into` -> `into_bound`, and `attach_borrow` -> `bind_borrowed`?

Reply:
Yes, that sounds consistent with the chosen names.